src/components/toggle/Toggle.tsx

```diff
@@ -1,105 +1,93 @@
 import cx from 'classnames';
 import omit from 'lodash.omit';
-import PropTypes from 'prop-types';
-import React, { createRef, PureComponent } from 'react';
+import React, { ChangeEvent, ReactNode, useRef } from 'react';
 import Box, { omitBoxProps, pickBoxProps } from '../box';
+import { BoxProps } from '../box/Box';
 import { TextBodyCompact, TextDisplay, TextSmall } from '../typography';
 import theme from './theme.css';
 
-/** @type {React.ComponentType<any>} */
-class Toggle extends PureComponent {
-  inputNode = createRef();
+type Size = 'small' | 'medium' | 'large';
 
-  handleToggle = (event) => {
-    const { disabled, onChange } = this.props;
+export interface ToggleProps extends Omit<BoxProps, 'className' | 'size'> {
+  checked?: boolean;
+  children?: ReactNode;
+  disabled?: boolean;
+  /** The maximum number of lines the label can take */
+  maxLines?: number;
+  name?: string;
+  className?: string;
+  label?: string;
+  onChange?: (event: ChangeEvent<HTMLInputElement>) => void;
+  size?: Size;
+}
 
-    if (event.pageX !== 0 && event.pageY !== 0) {
-      this.blur();
+const Toggle = ({
+  checked = false,
+  children,
+  disabled = false,
+  maxLines,
+  className,
+  label,
+  onChange,
+  size = 'medium',
+  ...others
+}: ToggleProps) => {
+  const ref = useRef<HTMLInputElement>(null);
+  const restProps = omit(others, ['onChange']);
```

Comment:
I don't think this `omit` does anything, `onChange` is already destructured on line 31